### ISSUE TITLE
sniper bots arent auto zooming within distance behind corners or walls

### ISSUE THREAD

**Issue by snipergaming134** [explowz/cathook#29]
yes auto-zoom & auto-rezoom is enabled
the max distance should be where its supposed to be around 1000

i have no clue what im doing wrong because the only time the snipers scope in is when you're out in the open and when they arent behind anything

plz help

**Comment by snipergaming888** [comment]
i am also having this issue lol
i thought it was my config and i set the scope distance to 5000, yet they still never scope in really.

**Comment by snipergaming888** [comment]
Video:

https://user-images.githubusercontent.com/81767390/236072828-e871053d-ba14-4f79-a04b-c703fa50232c.mp4

**Comment by snipergaming888** [comment]
I am using a clone of your rep, before you changed how getting players are handled.

**Comment by snipergaming134** [comment]
> Which catbot-setup repository are you using? If you're cloning the one from nullworks, then you're using their version of cathook. Bots work fine for me, and if there would be a problem with pre-zooming, then they should not zoom at all, since it'd basically mean that there's a problem detecting entities.

its happening with nullworks and your version, i have not messed with the distance at first, then after that i did to see if it fixed anything and nothing i did worked

**Comment by snipergaming134** [comment]
it never pre-scopes meaning it doesn't detect anybody near and only detects once it actually legit locks onto a player

**Comment by snipergaming134** [comment]
i have tried with many upon many configs and they all turned out the same exact way as each other

**Comment by snipergaming134** [comment]
i have a friend who doesn't have this problem and has the same settings set, while mine doesn't for some reason

**Comment by snipergaming134** [comment]
> i am also having this issue lol i thought it was my config and i set the scope distance to 5000, yet they still never scope in really. Edit: did some testing in a local server, it seems to be fixed if you set it to a high number (like 5000) EDIT EDIT: Found the issue. it seems zoom distance doesn't work at all with catbots, but the GUI works just fine. This has been confirmed in-game.

how do you use gui bots?

**Comment by snipergaming888** [comment]
> > i am also having this issue lol i thought it was my config and i set the scope distance to 5000, yet they still never scope in really. Edit: did some testing in a local server, it seems to be fixed if you set it to a high number (like 5000) EDIT EDIT: Found the issue. it seems zoom distance doesn't work at all with catbots, but the GUI works just fine. This has been confirmed in-game.
> 
> how do you use gui bots?

just launch the game normaly then inject cathook and load your config.

**Comment by snipergaming888** [comment]
> i have a friend who doesn't have this problem and has the same settings set, while mine doesn't for some reason

same, i know someone who is not having this problem

**Comment by snipergaming134** [comment]
> > > i am also having this issue lol i thought it was my config and i set the scope distance to 5000, yet they still never scope in really. Edit: did some testing in a local server, it seems to be fixed if you set it to a high number (like 5000) EDIT EDIT: Found the issue. it seems zoom distance doesn't work at all with catbots, but the GUI works just fine. This has been confirmed in-game.
> > 
> > 
> > how do you use gui bots?
> 
> just launch the game normaly then inject cathook and load your config.

arent you the owner of the sydney bots on badwater? if yes yours were scoping in :(

does that mean you own millions of computers at once lmao

**Comment by snipergaming888** [comment]
> > > > i am also having this issue lol i thought it was my config and i set the scope distance to 5000, yet they still never scope in really. Edit: did some testing in a local server, it seems to be fixed if you set it to a high number (like 5000) EDIT EDIT: Found the issue. it seems zoom distance doesn't work at all with catbots, but the GUI works just fine. This has been confirmed in-game.
> > > 
> > > 
> > > how do you use gui bots?
> > 
> > 
> > just launch the game normaly then inject cathook and load your config.
> 
> arent you the owner of the sydney bots on badwater? if yes yours were scoping in :(
> 
> does that mean you own millions of computers at once lmao

yes, but mine only scope in when they see people, thats the problem.

**Comment by snipergaming134** [comment]
actually ive noticed yours pre scoping

**Comment by snipergaming888** [comment]
> actually ive noticed yours pre scoping

when?

**Comment by snipergaming134** [comment]
on payload badwater while they were pushing the cart, i saw your bots kill mine before mine even reacted

**Comment by snipergaming888** [comment]
> on payload badwater while they were pushing the cart, i saw your bots kill mine before mine even reacted

I just tested them, they don't prescope.

**Comment by snipergaming134** [comment]
they didn't scope because they were already scoped in at that point and there was nobody else around

**Comment by snipergaming134** [comment]
> > on payload badwater while they were pushing the cart, i saw your bots kill mine before mine even reacted
> 
> I just tested them, they don't prescope.

weird-

**Comment by snipergaming134** [comment]
maybe i saw it wrong i dunno

**Comment by snipergaming888** [comment]
This is what they should be doing: https://www.youtube.com/watch?v=COfidnTbF1c&feature=youtu.be
Yet, they just walk around unscoped until they see someone.

**Comment by snipergaming134** [comment]
> This is what they should be doing: https://www.youtube.com/watch?v=COfidnTbF1c&feature=youtu.be Yet, they just walk around unscoped until they see someone. maybe i am Wrong, but imma do some more testing.

thats exactly what my issue is

**Comment by snipergaming134** [comment]
but i keep being told that its just my settings are wrong but theres no way thats the case

**Comment by snipergaming134** [comment]
also in that video were the bots gui because mine is ipc

**Comment by snipergaming888** [comment]
> also in that video were the bots gui because mine is ipc

Yeah

**Comment by snipergaming134** [comment]
can the ipc part be fixed or is that something that cant be?

**Comment by snipergaming888** [comment]
> can the ipc part be fixed or is that something that cant be?

IDK, i am trying to see what the issue is.

**Comment by snipergaming134** [comment]
oh ok

> > can the ipc part be fixed or is that something that cant be?
> 
> IDK, I am trying to see what the issue is.

oh ok

**Comment by snipergaming888** [comment]
plz fix explowz i lose every hvh without it working

**Comment by snipergaming888** [comment]
Tested 5/7/2023: new config, loaded both on GUI and bot, issue still persists.

**Comment by snipergaming888** [comment]
This has been fixed. you must put explowz cathook in your catbot-setup, compile it, then do sudo rm -rf /opt/cathook/bin, then do ./install-catbots.
